nipype/pipeline/engine/nodes.py

```diff
@@ -474,7 +474,9 @@ def run(self, updatehash=False):
         except Exception:
             logger.warning('[Node] Error on "%s" (%s)', self.fullname, outdir)
             # Tear-up after error
-            os.remove(hashfile_unfinished)
+            if not silentrm(hashfile_unfinished):
+                logger.debug('Unfinished hashfile %s does not exist',
+                             hashfile_unfinished)
```

Comment:
If this is likely to result from a race, would it be a good idea to flag that?

Reply:
Indeed, and maybe make a warning instead?